numpy/random/mtrand/mtrand.pyx

```diff
@@ -988,9 +988,9 @@ cdef class RandomState:
             raise ValueError("low is out of bounds for %s" % dtype)
         if ihigh > highbnd:
             raise ValueError("high is out of bounds for %s" % dtype)
-        if ilow >= ihigh:
-            raise ValueError("low >= high")
-
+        if ilow >= ihigh and np.prod(size) != 0:
+            raise ValueError("Range cannot be empty (low >= high) unless no samples are taken")
```

Comment:
Was suprised for a bit here, but I guess we do it like a python `range` and allow strange ranges as empty ranges, seems fine to me.